packages/url/src/filter-url-for-display.js

```diff
@@ -53,3 +53,19 @@ export function filterURLForDisplay( url, maxLength = null ) {
 		truncatedFile
 	);
 }
+
+/**
+ * Removes domain prefixes from a URL.
+ *
+ * @param {string} url The URL to strip domain prefixes from.
+ *
+ * @return {string} The URL without domain prefixes.
+ */
+export function stripDomainPrefixes( url ) {
+	if ( ! url ) {
+		return '';
+	}
+	return url
+		.replace( /^[a-z\-.\+]+[0-9]*:(\/\/)?/i, '' )
```

Comment:
Yes it makes sense to remove the fragment, query args and port too from the link. Just wanted to confirm, should I change the regex accordingly?

Reply:
> If this is for display purposes and link preview, it feels odd that we remove the protocol and the prefix, but we don't remove the `#hash` portion of the URL. And I can say the same about the port, and the query args.

@tyxla I'd totally agree. Wouldn't that be better addressed in the scope of https://github.com/WordPress/gutenberg/issues/61158 though? Your expertise would greatly help there to provide new utility functions for displaying purposes. I feel like this PR is going a little out of scope, as the original intent here was only to avoid to show duplicate information in the link control preview.

Reply:
> Yes it makes sense to remove the fragment, query args and port too from the link. Just wanted to confirm, should I change the regex accordingly?

Yes, I think it's worth considering how to name that helper function though, in order to reflect everything that it does. Technically, we're talking about returning the `hostname` (see [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL/hostname)), so the function could be called `getHostname()` or something similar.

> @tyxla I'd totally agree. Wouldn't that be better addressed in the scope of https://github.com/WordPress/gutenberg/issues/61158 though? Your expertise would greatly help there to provide new utility functions for displaying purposes. I feel like this PR is going a little out of scope, as the original intent here was only to avoid to show duplicate information in the link control preview.

To be fair, I don't mind where it will be addressed, but if this PR introduces a new API to a package, we need to ensure that it makes sense and is a reasonable API, and I was suggesting ways to get there. I'm happy to help with reviewing an alternative fix and handling this separately. If you feel like this PR is going too far, feel free to limit the fix to the affected component, avoiding introducing a new API, and that should make it more straightforward. We can then introduce the new function in a separate PR, if necessary.